### DISCUSSION TITLE
Limit Notes result by administrative boundary?

### DISCUSSION THREAD

**Discussion by stephan75** [ENT8R/NotesReview#104] [Ideas]
Whithout having searched whether there has already been an issuel / feature request like this:

So far it is possible to limit the Notes search result by the current map view bounding box.

Would it be possible to display only notes that are inside a certain OSM boundary relation?

Like inside Gemeinde Buxtehude, Landkreis Osterholz-Scharmbeck or Hamburg-Blankenese, to name but a few examples in Germany?

Would it be quite hard or more easy to implement such a feature in NotesReview in general?

**Comment by ENT8R** [comment]
This was already proposed in #100, even though the author of this ticket requests to filter by country only, if I implement this I will very likely not restrict it to selecting single countries but rather allow the user to specify their own desired boundaries, similar to [how OSMCha handles this](https://osmcha.org/filters).
One of the main tasks I need to do before implementing this is to find out how good the performance of the database is when users can input arbitrary (complex) boundaries.

**Comment by stephan75** [comment]
I understand ...

So what about the following ideas for a FIRST workflow model:

Have a look at https://geojson.io where you can draw rectangles and polygons on a map, and get a json file with its coordinates.

Implement such a drawing feature for Notes review on a slippy map.

thus you obtain a SIMPLE boundary for the beginning,

and then try a filtering with that to have all notes inside that square or polygon.

(I have NO knowledge about the database things that would be needed for such a task.)

When this first model works well, we could examine the performance when filtering geometries get more complex.

**Comment by ENT8R** [comment]
BTW, a few months ago I finally implemented this functionality, it is now possible to restrict the results to selected countries or custom user-defined shapes:

![screenrecording](https://github.com/ENT8R/NotesReview/assets/25306497/8bcb9bb5-cd15-441e-994e-b6e79c0e1435)
